### ISSUE TITLE
Crash when ejecting tape during cassette loading

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1242]
When ejecting a tape while the MSX is loading from it, you can get a crash:
```

Thread 1 "openmsx" received signal SIGSEGV, Segmentation fault.
openmsx::CassettePlayer::updateTapePosition (this=this@entry=0x555555db2e70, duration=duration@entry=..., time=...)
    at /usr/include/c++/9/bits/unique_ptr.h:360
360	      get() const noexcept
(gdb) bt full
#0  openmsx::CassettePlayer::updateTapePosition (this=this@entry=0x555555db2e70, duration=duration@entry=..., time=...)
    at /usr/include/c++/9/bits/unique_ptr.h:360
        __PRETTY_FUNCTION__ = "void openmsx::CassettePlayer::updateTapePosition(param, param)"
#1  0x0000555555665d3e in openmsx::CassettePlayer::sync (this=this@entry=0x555555db2e70, time=...) at src/cassette/CassettePlayer.cc:441
        duration = <optimized out>
#2  0x00005555556663d1 in openmsx::CassettePlayer::setState (this=0x555555db2e70, newState=openmsx::CassettePlayer::STOP, newImage=..., 
    time=...) at src/cassette/CassettePlayer.cc:247
        oldState = <optimized out>
        __PRETTY_FUNCTION__ = "void openmsx::CassettePlayer::setState(openmsx::CassettePlayer::State, const openmsx::Filename&, param)"
#3  0x0000555555666754 in openmsx::CassettePlayer::removeTape (this=0x555555db2e70, time=...) at /usr/include/c++/9/bits/char_traits.h:300
No locals.
#4  0x0000555555667d67 in openmsx::CassettePlayer::TapeCommand::execute (this=<optimized out>, tokens=..., result=..., time=...)
    at src/cassette/CassettePlayer.cc:695
        cassettePlayer = <optimized out>
#5  0x00005555557a5ed9 in openmsx::StateChangeDistributor::distribute (this=0x555555c97460, 
    event=std::shared_ptr<openmsx::StateChange> (use count 2, weak count 0) = {...}) at src/input/StateChangeDistributor.cc:72
        l = @0x555556320b78: 0x555555db3548
Python Exception <class 'gdb.error'> value has been optimized out: 
        __for_range = <synthetic pointer>
        __for_begin = <optimized out>
        __for_end = <optimized out>
        copy = <optimized out>
#6  0x00005555557a3ecc in openmsx::RecordedCommand::execute (this=0x555555db3508, tokens=..., result=...)
    at /usr/include/c++/9/bits/shared_ptr_base.h:756
        sa = {var = @0x555555db3568, oldValue = 0x555555db3560}
        time = <optimized out>
#7  0x000055555567e432 in openmsx::ProxyCmd::execute (this=<optimized out>, tokens=..., result=...) at src/commands/ProxyCommand.cc:38
        command = 0x555555db3508
#8  0x000055555567c4c1 in openmsx::Interpreter::commandProc (clientData=0x555555d99c30, interp=0x555555cb8be0, objc=2, objv=0x555555cc64e0)
    at src/commands/Interpreter.cc:179
        command = @0x555555d99c30: {<openmsx::CommandCompleter> = {<openmsx::Completer> = {
              _vptr.Completer = 0x555555bd7af8 <vtable for openmsx::ProxyCmd+16>, theName = "cassetteplayer", static output = 0x0}, 
            commandController = @0x555555c7b890}, allowInEmptyMachine = true, token = 0x555555d7c460}
        tokens = {static extent = <optimized out>, storage = {ptr = 0x555555cc64e0, size = 2}}
        res = 0
        result = {obj = 0x5555569ea470}
        __PRETTY_FUNCTION__ = "static int openmsx::Interpreter::commandProc(ClientData, Tcl_Interp*, int, Tcl_Obj* const*)"
#9  0x00007ffff7bf9f72 in TclNRRunCallbacks () from /usr/lib/x86_64-linux-gnu/libtcl8.6.so

```

**Comment by MBilderbeek** [comment]
This fix breaks the cassette hardware events, which are used by Catapult.
@m9710797 help :)

**Comment by MBilderbeek** [comment]
I analysed in more detail what is going on.

After this change, openMSX sends a media update event for the cassetteplayer, but it didn't even say yet that there is a cassetteplayer...

When switching to a different machine in Catapult, before this change was made, there were NO single update events for media. But after this change, there is:

```
UPDATE: connector, cassetteport, machine2, add
send 240: plug cassetteport
UPDATE: setting, autoruncassettes, machine2, true
UPDATE: media, cassetteplayer, machine2, 
Traceback (most recent call last):
```
And before this change:
```
UPDATE: connector, cassetteport, machine2, add
send 240: plug cassetteport
UPDATE: setting, autoruncassettes, machine2, true
UPDATE: setting, cassetteplayer_volume, machine2, 75
setting cassetteplayer_volume not registered
UPDATE: setting, cassetteplayer_balance, machine2, 0
setting cassetteplayer_balance not registered
UPDATE: setting, cassetteplayer_ch1_record, machine2, 
setting cassetteplayer_ch1_record not registered
UPDATE: setting, cassetteplayer_ch1_mute, machine2, false
setting cassetteplayer_ch1_mute not registered
UPDATE: sounddevice, cassetteplayer, machine2, add
send 241: set machine2::cassetteplayer_volume
send 242: set machine2::cassetteplayer_balance
send 243: openmsx_info setting machine2::cassetteplayer_balance
send 244: openmsx_info setting machine2::cassetteplayer_volume
send 245: openmsx_info setting machine2::cassetteplayer_volume
UPDATE: hardware, cassetteplayer, machine2, add
Adding media slot to admin admin for machine with id  machine2  and slot name  cassetteplayer
send 246: cassetteplayer
send 247: cassetteplayer
```
etc. Right near the end there, openMSX is telling Catapult that the cassetteplayer exists as a media slot.

**Comment by MBilderbeek** [comment]
m9710797 fixed this in 64ef0812e7926f3c4268ea085f35b54d58df3008. Thanks!
